choreolib/src/main/native/cpp/choreo/lib/ChoreoTrajectory.cpp

```diff
@@ -0,0 +1,103 @@
+#include "choreo/lib/ChoreoTrajectory.h"
+
+#include <wpi/json.h>
+
+using namespace choreolib;
+
+ChoreoTrajectory::ChoreoTrajectory(
+		const std::vector<ChoreoTrajectoryState> &states) : samples(states) {
+}
+
+ChoreoTrajectoryState ChoreoTrajectory::SampleInternal(
+		units::second_t timestamp) {
+	if (timestamp < samples[0].timestamp) {
```

Comment:
There's a more modern way to do this: https://github.com/wpilibsuite/allwpilib/blob/main/wpimath/src/main/native/cpp/trajectory/Trajectory.cpp#L66